rskj-core/src/main/java/co/rsk/core/bc/BlockUtils.java

```diff
@@ -35,20 +36,15 @@
     private static final Logger logger = LoggerFactory.getLogger("blockprocessor");
 
     public static boolean blockInSomeBlockChain(Block block, Blockchain blockChain) {
-        return blockInSomeBlockChain(block.getHash(), blockChain);
+        return blockInSomeBlockChain(block.getHash(), block.getNumber(), blockChain);
     }
 
-    public static boolean blockInSomeBlockChain(byte[] blockHash, Blockchain blockChain) {
-        final Block block = blockChain.getBlockByHash(blockHash);
+    public static boolean blockInSomeBlockChain(byte[] blockHash, long blockNumber, Blockchain blockChain) {
+        final ByteArrayWrapper key = new ByteArrayWrapper(blockHash);
+        final List<BlockInformation> blocks = blockChain.getBlocksInformationByNumber(blockNumber);
 
-        if (block == null)
-            return false;
-
-        final ByteArrayWrapper key = new ByteArrayWrapper(block.getHash());
-        final List<Block> blocks = blockChain.getBlocksByNumber(block.getNumber());
-
-        for (final Block b : blocks) {
-            if (new ByteArrayWrapper(b.getHash()).equals(key)) {
+        for (final BlockInformation bi : blocks) {
+            if (new ByteArrayWrapper(bi.getHash()).equals(key)) {
```

Comment:
ByteArrayWrapper unwraps the array when calling equals. It'd be nice to make a new method on the class called something like "equalsToByteArray(byte[] array)" that would do the FastByteComparisons.compareTo() and clal it from equals later.